Data/ByteString/Short/Internal.hs

```diff
@@ -195,11 +198,12 @@ import Foreign.C.String
   ( CString
   , CStringLen
   )
+#if !MIN_VERSION_base(4,11,0) && !PURE_HASKELL
```

Comment:
* `MIN_VERSION_base(4,11,0)` is pretty mysterious. I think this is really `HS_compareByteArrays_PRIMOP_AVAILABLE`?
* We don't have a job testing `base < 4.11` and `-fpure-haskell`. And now that I've thought to try it I see that it doesn't get this far anyway because `Word8#` doesn't exist with old GHCs.

  Well, I really don't care if `-fpure-haskell` works for old GHCs; perhaps just demand `base >= 4.16` when `pure-haskell` is set.

Reply:
> perhaps just demand `base >= 4.18` when pure-haskell is set.

+1.